clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/PartitionConsumptionState.java

```diff
@@ -1013,4 +1042,28 @@ public long getReadyToServeTimeLagThresholdInMs() {
   public void setReadyToServeTimeLagThresholdInMs(long readyToServeTimeLagThresholdInMs) {
     this.readyToServeTimeLagThresholdInMs = readyToServeTimeLagThresholdInMs;
   }
+
+  public void enableKeyUrnCompressionUponStartOfPush(List<String> keyUrnFields) {
+    if (keyUrnCompressor == null) {
+      LOGGER.info("Enabling key urn compression for replicaId: {} with fields: {}", replicaId, keyUrnFields);
+    } else {
+      LOGGER.info("Previous key urn compression dict will be overridden for replicaId: {}", replicaId);
+    }
+    this.keyUrnCompressor = new KeyUrnCompressor(keySchema, keyUrnFields, UrnDictV1.loadDict(Collections.emptyMap()));
+  }
+
+  public KeyUrnCompressionDict getKeyUrnCompressionDict() {
+    if (keyUrnCompressor == null) {
+      return null;
+    }
+    KeyUrnCompressionDict dict = new KeyUrnCompressionDict();
+    dict.keyUrnCompressionDictionaryVersion = 1;
```

Comment:
not an issue, but just a minor question though: what's the intention of the versioning here? Is it for that we want to implement a separate impl of the dictionary? I guess we will need to have extra handling to maintain the compatibility if that is the case.

Reply:
The dictionary schema evolution will be explicit by bumping the versions and no backward compatibility is needed.